packages/core/src/interpreter.ts

```diff
@@ -486,6 +490,13 @@ export class Interpreter<
       return;
     }
 
+    if (((event as EventObject)?.type || '').includes('error')) {
```

Comment:
it feels like this should be performed (if it needs to be done in the first place) after converting the `event` to `_event` (just a couple of lines below this)

We should also always test this with `startsWith`-like logic - because `some_error_event` should not be treated as an error. This logic is reserved for event types starting with `errror.`

Reply:
I refactored this part too, thank you